backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+    }
+
+    if (app.getAuthServerHost() != null && app.getAuthServerPort() != null) {
+      ManagedChannel authServiceChannel =
+          ManagedChannelBuilder.forTarget(app.getAuthServerHost() + ":" + app.getAuthServerPort())
+              .usePlaintext()
+              .intercept(authClientInterceptor.getClient1AuthInterceptor())
+              .build();
+      // all service stubs
+      organizationServiceBlockingStub = OrganizationServiceGrpc.newBlockingStub(authServiceChannel);
+      roleServiceBlockingStub = RoleServiceGrpc.newBlockingStub(authServiceChannel);
+      collaboratorServiceStub = CollaboratorServiceGrpc.newBlockingStub(authServiceChannel);
+    }
+
+    serverBuilder.build().start();
+    ManagedChannel channel = client1ChannelBuilder.maxInboundMessageSize(1024).build();
+    ManagedChannel client2Channel = client2ChannelBuilder.maxInboundMessageSize(1024).build();
+    deleteEntitiesCron = new DeleteEntitiesCron(authService, roleService, 1000);
+
+    // Create all service blocking stub
+    projectServiceStub = ProjectServiceGrpc.newBlockingStub(channel);
+    client2ProjectServiceStub = ProjectServiceGrpc.newBlockingStub(client2Channel);
+    datasetServiceStub = DatasetServiceGrpc.newBlockingStub(channel);
+    client2DatasetServiceStub = DatasetServiceGrpc.newBlockingStub(client2Channel);
+    repositoryServiceStub = VersioningServiceGrpc.newBlockingStub(channel);
+    client2RepositoryServiceStub = VersioningServiceGrpc.newBlockingStub(client2Channel);
+  }
+
+  @AfterClass
+  public static void removeServerAndService() {
+    App.initiateShutdown(0);
+
+    // Remove all entities
+    // removeEntities();
+    // Delete entities by cron job
+    deleteEntitiesCron.run();
+
+    // shutdown test server
+    serverBuilder.build().shutdownNow();
+  }
+
+  private static final Collection<Object[]> requestData =
+      Arrays.asList(
+          new Object[][] {
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_WRITE),
+              true,
+              true
+            },
+            {ResourceVisibility.ORG_DEFAULT, CollaboratorPermissions.newBuilder(), true, false},
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_ONLY),
+              true,
+              false
+            },
+            {ResourceVisibility.PRIVATE, CollaboratorPermissions.newBuilder(), false, false}
+          });
+
+  @Test
+  public void createProjectWithGlobalSharingOrganization() {
+    LOGGER.info("Global organization Project test start................................");
+
+    if (app.getAuthServerHost() == null || app.getAuthServerPort() == null) {
+      Assert.assertTrue(true);
+      return;
+    }
+
+    String orgName = "Org-test-verta";
```

Comment:
Does the organization already exist? If so, should this be a global constant?

Reply:
it doesn't exist